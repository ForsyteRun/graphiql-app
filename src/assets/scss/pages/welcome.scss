.welcome-container {
  max-width: 1440px; //TODO
  @include flexColumn(space-between);
  gap: 10rem;
  padding: 0 20px;
}

.welcome-title {
  align-self: flex-start;
}
