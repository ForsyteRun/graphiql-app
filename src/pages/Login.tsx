import { NavLink } from 'react-router-dom';
import { REGISTRATION_ROUTE } from '../constants/route';
import { LoginForm } from '../components';

const Login = () => {
  return (
    <>
      <div className="main page container">
        <div className="authorization">
          <div className="authorization__container">
            <h1>Войти</h1>
            <LoginForm />
          </div>
          <div className="authorization__information">
            <h4>Не зарегистрированы?</h4>
            <p>Создание учетной записи займет не больше минуты.</p>
            <NavLink to={REGISTRATION_ROUTE}>Создать учетную запись</NavLink>
          </div>
        </div>
      </div>
    </>
  );
};

export default Login;
