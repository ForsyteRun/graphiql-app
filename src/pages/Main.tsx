// Main.tsx

import GraphiQL from '../components/GraphiQL';
import { Footer } from '../layout/Footer';
import { Header } from '../layout/Header';

const Main = () => {
  return (
    <div className="tygh">
      <Header />
      <GraphiQL />
      <Footer />
    </div>
  );
};

export { Main };
