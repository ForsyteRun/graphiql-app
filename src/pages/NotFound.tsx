import { NavLink } from 'react-router-dom';
import { WELCOME_ROUTE } from '../constants/route';

const NotFound = () => {
  return (
    <div className="main container">
      <div className="exception">
        <div className="exception__code">
          404
          <span className="exception__code-txt">Ошибка</span>
        </div>
        <div className="exception__title-info">
          <h1 className="exception__title">
            Извините! Мы не смогли найти то, что вы искали.
          </h1>
          <p className="exception__info">Запрашиваемая страница не найдена.</p>
          <div className="exception__links">
            <div className="exception__links-item">
              <NavLink
                to={WELCOME_ROUTE}
                title=""
                className="exception__links-a"
              >
                Перейти на главную страницу
              </NavLink>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default NotFound;
