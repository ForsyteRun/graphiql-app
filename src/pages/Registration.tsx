import { NavLink } from 'react-router-dom';
import { RegistrationForm } from '../components';
import { AUTH_ROUTE } from '../constants/route';

const Registration = () => {
  return (
    <div className="tygh">
      <main className="main container page">
        <div className="authorization">
          <div className="authorization__container">
            <h1>Создать учетную запись</h1>
            <RegistrationForm />
          </div>
          <div className="authorization__information">
            <h4>Преимущества зарегистрированного пользователя:</h4>
            <div className="authorization__login">
              Уже есть учетная запись? Тогда просто жмите
              <NavLink to={AUTH_ROUTE}> Вход</NavLink>.
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

export default Registration;
