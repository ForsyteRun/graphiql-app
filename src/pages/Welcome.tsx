import { Content, OurTeam } from '../components';

const Welcome = () => {
  return (
    <div className="welcome-container">
      <div className="welcome-title">
        <h1>GraphiQL is the ability to harness the full power of GraphQL</h1>
        <span>make a request to your own universe</span>
      </div>
      <Content />
      <OurTeam />
    </div>
  );
};

export default Welcome;
