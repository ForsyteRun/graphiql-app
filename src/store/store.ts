import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import userReducer from './slice/userSlice';

const reducer = combineReducers({
  user: userReducer,
});

const store = configureStore({
  reducer,
});

setupListeners(store.dispatch);

export default store;
