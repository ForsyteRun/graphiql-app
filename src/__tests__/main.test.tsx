import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store/store';
import { Main } from '../pages';

const intersectionObserverMock = () => ({
  observe: () => null,
  disconnect: jest.fn(),
  unobserve: jest.fn(),
});

window.IntersectionObserver = jest
  .fn()
  .mockImplementation(intersectionObserverMock);

describe('main component', () => {
  test('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Main />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
