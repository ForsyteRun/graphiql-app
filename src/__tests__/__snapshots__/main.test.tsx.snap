// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container main"
  >
    <h1
      class="main__title"
    >
      GraphiQL Editor
    </h1>
    <div
      class="api"
    >
      <div
        class="api__content"
      >
        <div
          class="api__label"
        >
          API:
        </div>
        <input
          class="api__input"
          name="api"
          placeholder="введите api"
          value="https://rickandmortyapi.com/graphql"
        />
      </div>
      <div
        class="button"
      >
        Submit
      </div>
    </div>
    <div
      class="headers"
    >
      <div
        class="headers__content"
      >
        <div
          class="headers__label"
        >
          Headers:
        </div>
        <input
          class="headers__input"
          name="headers"
          value=""
        />
      </div>
      <div
        class="button"
      >
        Submit
      </div>
    </div>
    <div
      class="variables"
    >
      <div
        class="variables__content"
      >
        <div
          class="variables__label"
        >
          Variables:
        </div>
        <input
          class="variables__input"
          name="variables"
          value=""
        />
      </div>
      <div
        class="button"
      >
        Submit
      </div>
    </div>
    <div
      class="editor"
    >
      <div
        class="editor__wrapper"
      >
        <div
          class="editor__left"
        >
          <div
            class="editor__query "
          />
        </div>
        <div
          class="editor__right"
        >
          <div
            class="editor__response"
          />
        </div>
      </div>
    </div>
    <aside
      class=""
    >
      <div
        class="docs-container"
      >
        <button
          style="border: 1px solid red;"
          type="button"
        >
          Go up
        </button>
        <ul
          class="fieldSchema-container"
        />
      </div>
      <button
        class="docs-button"
        disabled=""
      >
        Docs
      </button>
    </aside>
  </div>
</DocumentFragment>
`;
